File: java-junit-gradle/src/test/java/varar/RunVararTest.java
package varar;

import org.junit.platform.suite.api.IncludeEngines;
import org.junit.platform.suite.api.SelectDirectories;
import org.junit.platform.suite.api.Suite;

/**
 * Maven Surefire and Gradle only discover class-based tests, so this {@code @Suite} is the
 * bridge that asks the JUnit Platform to run the {@code "varar"} engine over the oath corpus.
 * varar.config.json (in this project's root, the test working directory) decides which .md
 * files are oaths and which classes define the steps. The {@code *Test} suffix matters:
 * Surefire only scans classes matching its naming convention.
 */
@Suite
@IncludeEngines("varar")
@SelectDirectories(".")
public class RunVararTest {}

File: kotlin-junit/src/test/kotlin/varar/RunVararTest.kt
package varar

import org.junit.platform.suite.api.IncludeEngines
import org.junit.platform.suite.api.SelectDirectories
import org.junit.platform.suite.api.Suite

// Maven Surefire and Gradle only discover class-based tests, so this @Suite is
// the bridge that asks the JUnit Platform to run the "varar" engine over the oath
// corpus (the .md files in this project). varar.config.json decides which .md
// files are oaths and which classes define the steps. The
// *Test suffix matters under Maven: Surefire only scans classes matching its
// naming convention.
@Suite @IncludeEngines("varar") @SelectDirectories(".") class RunVararTest
